input.py

```diff
@@ -1,223 +1,261 @@
+    entries = []
+    for entry in input_dir.iterdir():
+        # Exclude hidden files/folders (starting with '.')
+        if entry.name.startswith('.'):
+            continue
+
+        # Check for directories or files with video extensions
+        is_video = entry.suffix.lower() in ('.mkv', '.mp4')
+        if entry.is_dir() or is_video:
+            # Append a trailing '/' for directories in the display list
+            display_name = entry.name + '/' if entry.is_dir() else entry.name
+            # Store (display_name, full_path)
+            entries.append((display_name, entry))
+
+    # Sort the display names for presentation
+    entries.sort(key=lambda x: x[0])
+
+    # Extract just the display names for the survey selection
+    display_entries = [d for d, p in entries]
+
+    # Map back from display name to the full Path object after selection
+    path_map = {d: p for d, p in entries}
+
+    # 2. Survey selection
+    selection = survey.routines.basket(
+        f'Select folders/files in {input_dir}:',
+        options=display_entries,
+        view_max=None
+    )
+    # Get the selected Path objects
+    selected_paths = [path_map[display_entries[i]] for i in selection]
+
+    # 3. Process selected items
+    files = []
+    full_files = []
+
+    for item_path in selected_paths:
+        if item_path.is_dir():
+            # Use Path.glob for a simple recursive search (**/*)
+            # or use Path.rglob() for cleaner code
+            for file_path in item_path.rglob('*'):
+                if file_path.suffix.lower() in ('.mp4', '.mkv'):
+                    # The file name only
+                    files.append(file_path.name)
+                    # The full absolute path string
+                    full_files.append(str(file_path.resolve()))
+        elif item_path.suffix.lower() in ('.mp4', '.mkv'):
+            # The file name only (it's already the file)
+            files.append(item_path.name)
+            # The full absolute path string
+            full_files.append(str(item_path.resolve()))
+
+    return files, full_files
+
+
+def parse_episode_code(episode_code: str):
+    """Return (season, episode) numbers from 'SxxExx', or (0, 0) if invalid."""
+    match = re.search(r"[Ss](\d+)[Ee](\d+)", episode_code or "")
+    if not match:
+        return 0, 0
+    return map(int, match.groups())
+
+
+# == MAIN LOGIC ==
+raw_id_or_title = survey.routines.input('ID or Title: ', value='').strip()
+if not raw_id_or_title:
+    # If no ID or title provided, prompt user to enter a TMDB ID.
+    print("No ID or title provided; please enter a TMDB ID.")
+    raw_id_or_title = survey.routines.input('ID or Title: ', value='').strip()
+
+mediatypes = ('tv', 'movie')
+defaultjob['type'] = mediatypes[survey.routines.select(
+    'Type: ', options=mediatypes)]
+ontmdb = False
+sname, syear = '', 0
+tmdbid = None
+
+def is_tmdb_id(text: str) -> bool:
+    return text.startswith('tmdb-') or text.isdigit()

```

Comment:
```suggestion
    return text.startswith('tmdb-')
```
This causes an issue for media that has a name which is just a number. For example the film "1917" will instead look for the ID and find "Who Killed Bambi?". I suggest solving this by only allowing tmdb ids by starting with "tmdb-" as if support is later added for other media databases (imdb) it won't conflict.